persistence/schema/gov.go

```diff
@@ -3,440 +3,151 @@ package schema
-		hex.EncodeToString(params.ServiceNodeMinimumPauseBlocksOwner),
-		hex.EncodeToString(params.ServiceNodeMaxPausedBlocksOwner),
-		hex.EncodeToString(params.ServiceNodesPerSessionOwner),
-		hex.EncodeToString(params.FishermanMinimumStakeOwner),
-		hex.EncodeToString(params.FishermanMaxChainsOwner),
-		hex.EncodeToString(params.FishermanUnstakingBlocksOwner),
-		hex.EncodeToString(params.FishermanMinimumPauseBlocksOwner),
-		hex.EncodeToString(params.FishermanMaxPausedBlocksOwner),
-		hex.EncodeToString(params.ValidatorMinimumStakeOwner),
-		hex.EncodeToString(params.ValidatorUnstakingBlocksOwner),
-		hex.EncodeToString(params.ValidatorMinimumPauseBlocksOwner),
-		hex.EncodeToString(params.ValidatorMaxPausedBlocksOwner),
-		hex.EncodeToString(params.ValidatorMaximumMissedBlocksOwner),
-		hex.EncodeToString(params.ValidatorMaxEvidenceAgeInBlocksOwner),
-		hex.EncodeToString(params.ProposerPercentageOfFeesOwner),
-		hex.EncodeToString(params.MissedBlocksBurnPercentageOwner),
-		hex.EncodeToString(params.DoubleSignBurnPercentageOwner),
+func SetParamQuery[T ParamTypes](paramName string, paramValue T, height int64) string {
+	//TODO (@deblasis): Fix this
+	fields := "name,height,value,enabled,type"
 
-		hex.EncodeToString(params.MessageDoubleSignFeeOwner),
-		hex.EncodeToString(params.MessageSendFeeOwner),
-		hex.EncodeToString(params.MessageStakeFishermanFeeOwner),
-		hex.EncodeToString(params.MessageEditStakeFishermanFeeOwner),
-		hex.EncodeToString(params.MessageUnstakeFishermanFeeOwner),
-		hex.EncodeToString(params.MessagePauseFishermanFeeOwner),
-		hex.EncodeToString(params.MessageUnpauseFishermanFeeOwner),
-		hex.EncodeToString(params.MessageFishermanPauseServiceNodeFeeOwner),
-		hex.EncodeToString(params.MessageTestScoreFeeOwner),
-		hex.EncodeToString(params.MessageProveTestScoreFeeOwner),
-		hex.EncodeToString(params.MessageStakeAppFeeOwner),
-		hex.EncodeToString(params.MessageEditStakeAppFeeOwner),
-		hex.EncodeToString(params.MessageUnstakeAppFeeOwner),
-		hex.EncodeToString(params.MessagePauseAppFeeOwner),
-		hex.EncodeToString(params.MessageUnpauseAppFeeOwner),
-		hex.EncodeToString(params.MessageStakeValidatorFeeOwner),
-		hex.EncodeToString(params.MessageEditStakeValidatorFeeOwner),
-		hex.EncodeToString(params.MessageUnstakeValidatorFeeOwner),
-		hex.EncodeToString(params.MessagePauseValidatorFeeOwner),
-		hex.EncodeToString(params.MessageUnpauseValidatorFeeOwner),
-		hex.EncodeToString(params.MessageStakeServiceNodeFeeOwner),
-		hex.EncodeToString(params.MessageEditStakeServiceNodeFeeOwner),
-		hex.EncodeToString(params.MessageUnstakeServiceNodeFeeOwner),
-		hex.EncodeToString(params.MessagePauseServiceNodeFeeOwner),
-		hex.EncodeToString(params.MessageUnpauseServiceNodeFeeOwner),
-		hex.EncodeToString(params.MessageChangeParameterFeeOwner),
-		DefaultBigInt,
-	)
-}
+	subQuery := fmt.Sprintf(`SELECT %s`, fields)
+	subQuery += fmt.Sprintf(` FROM %s WHERE name='%s' AND height=%d`, ParamsTableName, paramName, height)
```

Comment:
For `Get`, we do `<=height` to get the latest value from the historical state.

For `set`, we want to use the height to set the value, no need to query on it.

For `nullify`, let's delete that altogether. It's legacy and we can add it back if we need to.
- For flags, we can turn them on/off
- For params, we can set the values to 0 or -1 via Set

This will keep the design simpler for now.